src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -361,5 +365,63 @@ private async Task FormatDocumentCreatedFromTemplateAsync(IVsHierarchy hierarchy
                 formattedText.Write(textWriter, cancellationToken: CancellationToken.None);
             });
         }
+
+        private static Project AddEditorConfigFiles(Project projectToAddTo, string projectFolder)
+        {
+            do
+            {
+                projectToAddTo = AddEditorConfigFile(projectToAddTo, projectFolder, out var foundRoot);
+
+                if (foundRoot)
+                {
+                    break;
+                }
+                projectFolder = Path.GetDirectoryName(projectFolder);
+            }
+            while (projectFolder is not null);
+
+            return projectToAddTo;
+
+            static Project AddEditorConfigFile(Project project, string folder, out bool foundRoot)
+            {
+                const string EditorConfigFileName = ".editorconfig";
+
+                foundRoot = false;
+
+                var editorConfigFile = Path.Combine(folder, EditorConfigFileName);
```

Comment:
any concern about Path.Combine throwing?

Reply:
Famous last words perhaps, but no. Path.Combine only throws if one of the paths is null or has invalid characters, but since one part is a constant, and the other part is the location where Visual Studio has already successfully written a file, it should be fine.

Reply:
Sgtm